### PR TITLE
Use `importjs`

### PR DESCRIPTION
**Pull request by konsumlamm**
This PR changes uses of `importcpp` to `importjs` where it doesn't make any problems.

I looked at all files with "js" in the path name, those that I didn't change are:

* `js/dom.nim`
* `js/dom_extensions.nim`
* `js/jsconsole.nim`
* `js/jscore.nim`
* `system/jssys.nim`

Using `importjs` there gives `` `importjs` for routines requires a pattern`` errors, since apparently `importjs` doesn't behave the same as `importc` and `importcpp`. Should I open an issue for this or just add patterns for the imports (in a future PR)?

### CODE REVIEW

**Review by dom96 — APPROVED** [review]
Nice! LGTM

File: lib/js/asyncjs.nim [hunk 1]
```diff
@@ -70,7 +70,7 @@ type
     future*: T
   ## Wraps the return type of an asynchronous procedure.
 
-  PromiseJs* {.importcpp: "Promise".} = ref object
+  PromiseJs* {.importjs: "Promise".} = ref object
```
Comment (timotheecour) [inline lib/js/asyncjs.nim:73]: see also https://github.com/timotheecour/Nim/issues/662

**Review by timotheecour — APPROVED** [review]
LGTM. the reason this now works is that docs for nim js modules are now built with the new `-b:js` flag

**Comment by timotheecour** [comment]
> Using importjs there gives `importjs` for routines requires a pattern errors, since apparently importjs doesn't behave the same as importc and importcpp. Should I open an issue for this or just add patterns for the imports (in a future PR)?

it should be discussed in https://github.com/nim-lang/RFCs/issues/315, please add a note there if this use case isn't covered already

**Pull request merged**